### ISSUE TITLE
Consider making `request-permission-to-use` aware of user activation

### ISSUE THREAD

**Issue by engedy** [w3c/permissions#194]
The [dfn-obtain-permission](https://w3c.github.io/wake-lock/#dfn-obtain-permission) algorithm could probably take an additional bit in some shape or form prescribing whether user activation is required for showing a permission prompt if the permission state is "prompt".

E.g., the Wake Lock spec currently works around this limitation by essentially re-implementing its own version of this algorithm in [obtain-permission](https://w3c.github.io/wake-lock/#dfn-obtain-permission).

**Comment by engedy** [comment]
@rakuco, FYI.

**Comment by jyasskin** [comment]
I think we're missing a lot of support for user activation in the permissions spec, partly because some Firefox folks weren't convinced that it made sense to gate prompts on user activation at all. (Although #107 proposed it for all prompts, not just a subset.)

@mustaqahmed, how's https://github.com/whatwg/html/pull/3851 going? I kinda think we should wait to explicitly support user activation in the Permissions spec until we have cross-browser agreement on what user activation is. I think I'd support making the "Ask the user’s permission" step give the UA a bit more choice until then, for example by saying "If the user is expected to be ready to think about a prompt, ask ...".

**Comment by mustaqahmed** [comment]
@jyasskin: We are waiting for support from other browsers to land https://github.com/whatwg/html/pull/3851.

**Comment by engedy** [comment]
@jyasskin, thank you, that's great context!

I am concerned that the Permissions API spec makes no mention of user activation whatsoever, while a number of other specs either normatively require [1,2] or at least encourage using [3] user gesture/action/activation as part of their permission flows. We need to create consistency around this. Who would be the key stakeholders from Firefox to involve in this discussion?

[1] https://w3c.github.io/clipboard-apis/#privacy-permission
[2] https://w3c.github.io/deviceorientation/#deviceorientation
[3] https://immersive-web.github.io/webxr/#applicationflow

**Comment by mustaqahmed** [comment]
FYI: The HTML spec [change](https://github.com/whatwg/html/pull/3851) landed yesterday.

**Comment by beaufortfrancois** [comment]
I'd love if we could make progress on this issue as this would allow specs like [WebNFC](https://github.com/w3c/web-nfc/pull/449) to simply use the Permissions API without adding an extra step for user activation.

**Comment by mustaqahmed** [comment]
Yes, fixing this issue could simplify [quite a few](https://docs.google.com/presentation/d/1ksUg-ug4FNHWviYdMlU0ATWfmiibLXEEV0btHaRg44s/edit#slide=id.g62553735ea_0_47) other specs by removing their _direct_ link to user activation, effectively making them indirectly connected to user activation through Permissions.

**Comment by dbaron** [comment]
I'd also note that #189 is somewhat related here, in that it proposes that the API should expose more information about whether user activation is needed.  (I think this issue is about the relationship of user activation to the permission requesting algorithm, and #189 is at least in significant part about the relationship of user activation to `permissions.query`.)

**Comment by beaufortfrancois** [comment]
@mustaqahmed @jyasskin Is there anything we can do to help move this change forward?

**Comment by reillyeon** [comment]
As an editor on the WebUSB and Web Bluetooth specifications I'm supportive of this proposal for the same reasons as @engedy's original comment. Enforcing user activation from inside a specification that depends on the Permissions API is currently very awkward.

**Comment by marcoscaceres** [comment]
I can't imagine any situation where the user would be prompted (i.e, UI is shown), but the user activation wouldn't be consumed (...but I may be lacking imagination, so someone enlighten me?). 

Thus, I think we should add these checks... It would require modifying the (or adding a new) algorithm to take a promise. The transient activation would be checked (potentially causing a promise rejection), user activation would be consumed, and the prompt would be shown. 

We could keep things backwards compatible by making the promise optional.

**Comment by marcoscaceres** [comment]
Sent Draft PR: https://github.com/w3c/permissions/pull/401
